src/System.Buffers/tests/ArrayPool/UnitTests.cs

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using Xunit;
+
+namespace System.Buffers.ArrayPool.Tests
+{
+    public partial class ArrayPoolUnitTests
+    {
+        /*
+            NOTE - due to test parallelism and sharing, use an instance pool for testing unless necessary
+        */ 
+        [Fact]
+        public static void SharedInstanceCreatesAnInstanceOnFirstCall()
+        {
+            Assert.True(ArrayPool<byte>.Shared != null);
+        }
+
+        [Fact]
+        public static void SharedInstanceOnlyCreatesOneInstanceOfOneTypep()
+        {
+            ArrayPool<byte> instance = ArrayPool<byte>.Shared;
+            Assert.True(instance == ArrayPool<byte>.Shared);
+        }
+
+        [Fact]
+        public static void CreateWillCreateMultipleInstancesOfTheSameType()
+        {
+            Assert.True(ArrayPool<byte>.Create() != ArrayPool<byte>.Create());
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-1)]
+        public static void CreatingAPoolWithInvalidArrayCountThrows(int length)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => ArrayPool<byte>.Create(numberOfArrays: length));
```

Comment:
Throws has an overload that takes a first argument that's the name of the parameter that was out of range.  We should specify the name so that it verifies it matches. (Same for other tests that use Throws with arg exceptions, but I won't comment on those.)